test/tools_tests/branch_length_estimator_tests/iid_exponential_mle_test.py

```diff
@@ -0,0 +1,363 @@
+                "4": [1, 0, 3, 0, 0, 0, 0, 0, 0, -1],
+                "5": [0, 0, 0, 0, 5, 6, 7, 0, 0, -1],
+                "6": [0, 0, 0, 4, 0, 6, 0, 8, 9, -1],
+            }
+        )
+        model = IIDExponentialMLE(minimum_branch_length=0.0, solver=solver)
+        model.estimate_branch_lengths(tree)
+        self.assertAlmostEqual(model.mutation_rate, 0.378, places=3)
+        self.assertAlmostEqual(
+            tree.get_branch_length("0", "1"), 0.537, places=3
+        )
+        self.assertAlmostEqual(
+            tree.get_branch_length("0", "2"), 0.219, places=3
+        )
+        self.assertAlmostEqual(
+            tree.get_branch_length("1", "3"), 0.463, places=3
+        )
+        self.assertAlmostEqual(
+            tree.get_branch_length("1", "4"), 0.463, places=3
+        )
+        self.assertAlmostEqual(
+            tree.get_branch_length("2", "5"), 0.781, places=3
+        )
+        self.assertAlmostEqual(
+            tree.get_branch_length("2", "6"), 0.781, places=3
+        )
+        self.assertAlmostEqual(model.log_likelihood, -22.689, places=3)
+
+    @parameterized.expand([("ECOS", "ECOS"), ("SCS", "SCS")])
+    def test_on_simulated_data(self, name, solver):
+        """
+        We run the estimator on data simulated under the correct model.
+        The estimator should be close to the ground truth.
+        """
+        tree = nx.DiGraph()
+        tree.add_nodes_from(["0", "1", "2", "3", "4", "5", "6"]),
+        tree.add_edges_from(
+            [
+                ("0", "1"),
+                ("0", "2"),
+                ("1", "3"),
+                ("1", "4"),
+                ("2", "5"),
+                ("2", "6"),
+            ]
+        )
+        tree = CassiopeiaTree(tree=tree)
+        tree.set_times(
+            {"0": 0, "1": 0.1, "2": 0.9, "3": 1.0, "4": 1.0, "5": 1.0, "6": 1.0}
+        )
+        np.random.seed(1)
+        Cas9LineageTracingDataSimulator(
+            number_of_cassettes=200,
+            size_of_cassette=1,
+            mutation_rate=1.5,
+        ).overlay_data(tree)
+        model = IIDExponentialMLE(minimum_branch_length=0.0, solver=solver)
+        model.estimate_branch_lengths(tree)
+        self.assertTrue(0.05 < tree.get_time("1") < 0.15)
+        self.assertTrue(0.8 < tree.get_time("2") < 1.0)
+        self.assertTrue(0.9 < tree.get_time("3") < 1.1)
+        self.assertTrue(0.9 < tree.get_time("4") < 1.1)
+        self.assertTrue(0.9 < tree.get_time("5") < 1.1)
+        self.assertTrue(0.9 < tree.get_time("6") < 1.1)
+        self.assertTrue(1.4 < model.mutation_rate < 1.6)
+        self.assertAlmostEqual(tree.get_time("0"), 0.0, places=3)
+
+    @parameterized.expand([("ECOS", "ECOS"), ("SCS", "SCS")])
+    def test_subtree_collapses_when_no_mutations(self, name, solver):
+        """
+        A subtree with no mutations should collapse to 0. It reduces the problem to
+        the same as in 'test_hand_solvable_problem_1'
```

Comment:
This test is failing for me:

```
self.assertAlmostEqual(model.log_likelihood, -1.386, places=3)
E   AssertionError: -inf != -1.386 within 3 places
```

Reply:
I should be more specific - it looks like it works with ECOS but fails with SCS.

Reply:
Yes, this is really weird. The test passes on the cluster (and on Github), but fails on my machine and on yours. I'll look into it.

Reply:
Could this be a difference in `cvxpy` versions? My version that is failing with is 1.1.7